pkg/prometheus/server/statefulset_test.go

```diff
@@ -1538,21 +1405,8 @@ func TestRetentionAndRetentionSize(t *testing.T) {
 			}
 		}
 
-		if foundRetention != test.shouldContainRetention || foundRetentionFlag != test.shouldContainRetention {
-			if test.shouldContainRetention {
-				t.Fatalf("test %d, expected Prometheus args to contain %v, but got %v", i, test.expectedRetentionArg, promArgs)
-			} else {
-				t.Fatalf("test %d, expected Prometheus args to NOT contain %v, but got %v", i, test.expectedRetentionArg, promArgs)
-			}
-		}
-
-		if foundRetentionSize != test.shouldContainRetentionSize || foundRetentionSizeFlag != test.shouldContainRetentionSize {
-			if test.shouldContainRetentionSize {
-				t.Fatalf("test %d, expected Prometheus args to contain %v, but got %v", i, test.expectedRetentionSizeArg, promArgs)
-			} else {
-				t.Fatalf("test %d, expected Prometheus args to NOT contain %v, but got %v", i, test.expectedRetentionSizeArg, promArgs)
-			}
-		}
+		require.False(t, (foundRetention != test.shouldContainRetention || foundRetentionFlag != test.shouldContainRetention), "test %d, expected Prometheus args to %scontain %v, but got %v", i, map[bool]string{true: "", false: "NOT "}[test.shouldContainRetention], test.expectedRetentionArg, promArgs)
+		require.False(t, (foundRetentionSize != test.shouldContainRetentionSize || foundRetentionSizeFlag != test.shouldContainRetentionSize), "test %d, expected Prometheus args to %scontain %v, but got %v", i, map[bool]string{true: "", false: "NOT "}[test.shouldContainRetentionSize], test.expectedRetentionSizeArg, promArgs)
```

Comment:
We don't necessarily need to keep the previous message with this refactoring. It would be even better to not add them at all I believe.

We added messages with t.Fatal because t.Fatal has no context whatsoever. The require package already adds a lot of context to the test, and emits nice human readable comparisons when the test fails.

We could simplify here with just:

```go
if test.shouldContainRetention {
      require.True(t, (foundRetention && foundRetentionFlag))
}

if test.shouldContainRetentionSize {
      require.True(t, (foundRetentionSize && foundRetentionSizeFlag))
}
```